python/tvm/micro/model_library_format.py

```diff
@@ -277,6 +294,37 @@ def _create_empty_entry(target_device_type):
             main_func_metadata.io_sizes[target]
         )
 
+        # Now, we also add the information about the size of each input and output of the main
+        # function (in bytes)
+        input_dict = {}
+        for input_param in main_func_metadata.relay_primfuncs[target].params:
+            if hasattr(input_param, "checked_type"):
+                input_dict[input_param.name_hint] = int(
+                    _shape_to_size(input_param.checked_type.shape, input_param.checked_type.dtype)
+                )
+            else:
+                # TODO: maybe fill checked_type here?
+                input_dict[input_param.name_hint] = 0
+        target_main_entries[int(target.kind.device_type)]["inputs"] = input_dict
+
+        output_dict = {}
+        # For output, we dont have the name of the output, so we enumerate them
+        if isinstance(main_func_metadata.relay_primfuncs[target].ret_type, tvm.ir.type.TupleType):
+            for i, output_type in enumerate(
+                main_func_metadata.relay_primfuncs[target].ret_type.fields
+            ):
+                if hasattr(output_type, "shape"):
+                    output_dict[i] = int(_shape_to_size(output_type.shape, output_type.dtype))
+                else:
+                    output_dict[i] = 0
+        else:
+            output_type = main_func_metadata.relay_primfuncs[target].ret_type
+            if hasattr(output_type, "shape"):
+                output_dict[0] = int(_shape_to_size(output_type.shape, output_type.dtype))
+            else:
+                output_dict[0] = 0
```

Comment:
This is duplicated between input and output, we can replace it with a function:

```python
def _create_type_metadata(input_type):
    return {
        "size": int(
            _shape_to_size(input_type.shape, input_type.dtype)
        ),
        "dtype": str(input_type.dtype)
    }
```

Reply:
Applied the proposed change